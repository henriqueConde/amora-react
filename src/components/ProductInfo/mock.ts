export default {
  title: 'Chanel',
  price: '215,00',
  description:
    'Experience the mysterious sent of chanel green ocean. Experience the mysterious sent of chanel green ocean.'
}
