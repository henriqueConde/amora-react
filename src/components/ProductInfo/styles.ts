import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    background: ${theme.colors.white};
    padding: ${theme.spacings.small};
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0, 0, 0, 0.2);
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray};
    margin-bottom: ${theme.spacings.small};
    ${media.greaterThan('medium')`
      max-width: 77rem;
    `}
  `}
`

export const ButtonsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    > button {
      width: 100%;
      margin-bottom: ${theme.spacings.xxsmall};
    }
    ${media.greaterThan('medium')`
      flex-direction: row-reverse;
      > button {
        width: initial;
        margin-bottom: 0;
      }
    `}
  `}
`
