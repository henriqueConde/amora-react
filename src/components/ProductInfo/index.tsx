import {
  AddShoppingCart,
  FavoriteBorder
} from '@styled-icons/material-outlined'

import Button from 'components/Button'
import Heading from 'components/Heading'

import * as S from './styles'

export type ProductInfoProps = {
  title: string
  description: string
}

const ProductInfo = ({ title, description }: ProductInfoProps) => (
  <S.Wrapper data-cy="product-info">
    <Heading color="black" lineBottom>
      {title}
    </Heading>

    <S.Description>{description}</S.Description>

    <S.ButtonsWrapper>
      <Button icon={<AddShoppingCart />} size="large">
        Add to cart
      </Button>
      <Button icon={<FavoriteBorder />} size="large" minimal>
        Wishlist
      </Button>
    </S.ButtonsWrapper>
  </S.Wrapper>
)

export default ProductInfo
