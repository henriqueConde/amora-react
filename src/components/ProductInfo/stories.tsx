import { Story, Meta } from '@storybook/react/types-6-0'
import ProductInfo, { ProductInfoProps } from '.'
import mockProduct from './mock'

export default {
  title: 'Product/ProductInfo',
  component: ProductInfo,
  parameters: {
    backgrounds: {
      default: 'amora-dark'
    }
  },
  args: mockProduct
} as Meta

export const Default: Story<ProductInfoProps> = (args) => (
  <div style={{ maxWidth: '144rem', margin: 'auto', padding: '1.5rem' }}>
    <ProductInfo {...args} />
  </div>
)
