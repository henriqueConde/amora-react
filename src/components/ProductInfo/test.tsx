import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import ProductInfo from '.'
const props = {
  title: 'My Product Title',
  description: 'Product Description',
  price: 215
}

describe('<ProductInfo />', () => {
  it('should render product informations', () => {
    const { container } = renderWithTheme(<ProductInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /my product title/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/product description/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render buttons', () => {
    renderWithTheme(<ProductInfo {...props} />)
    expect(
      screen.getByRole('button', { name: /add to cart/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /wishlist/i })
    ).toBeInTheDocument()
  })
})
