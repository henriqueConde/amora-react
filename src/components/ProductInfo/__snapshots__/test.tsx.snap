// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo /> should render product informations 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #038FC9;
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c4:hover {
  background: #026b97;
}

.c4 svg {
  width: 1.5rem;
}

.c4 svg + span {
  margin-left: 0.8rem;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #038FC9;
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: none;
  color: #038FC9;
}

.c6:hover {
  background: #026b97;
}

.c6 svg {
  width: 1.5rem;
}

.c6 svg + span {
  margin-left: 0.8rem;
}

.c6:hover {
  color: #026b97;
}

.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #038FC9;
}

.c0 {
  position: relative;
  background: #FAFAFA;
  padding: 2.4rem;
  box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
}

.c2 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > button {
  width: 100%;
  margin-bottom: 0.8rem;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c2 {
    max-width: 77rem;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .c3 > button {
    width: initial;
    margin-bottom: 0;
  }
}

<div
  class="c0"
  data-cy="product-info"
>
  <h2
    class="c1"
    color="black"
  >
    My Product Title
  </h2>
  <p
    class="c2"
  >
    Product Description
  </p>
  <div
    class="c3"
  >
    <button
      class="c4"
    >
      <svg
        aria-hidden="true"
        class="c5"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c5"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span>
        Wishlist
      </span>
    </button>
  </div>
</div>
`
